### PR TITLE
Fixes #2121 - Teleporting to a home/spawn that is in water from another world puts you in the wrong place.

### LINKED ISSUE #2121: (1.13) Teleporting to a home/spawn that is in water from another world puts you in the wrong place.
Within the same world, you get put in the water at the right coordinates. 

From another world, you get put in the nearest air block above the X, Z coordinate.

Re-sending the command will put you in the proper place.

### PR DESCRIPTION
**Pull request by caojohnny**
Fixes #2121.

Prior to the addition of this patch, teleporting from another world through commands such as `/spawn` and `/home` would cause players to be teleported to the surface of the water. After this patch, using the same command will correctly teleport them to the original location.

In seeing that the addition of water would cause the `HOLLOW_MATERIALS` set to be identical to the `TRANSPARENT_MATERIALS`, I have removed the latter's usage and simply added water to the former.

I'm not exactly sure if adding water to `HOLLOW_MATERIALS` is the right decision, but it fixes the issue, and I personally don't really see any point in not having water in the list. I imagine some people might use this as a way to drop players on the surface of the water, but they can fix that issue quite easily by actually going to the surface and setting the location there. I also can see that water is not necessarily a "safe" location because players can drown, but I really see no other alternative. 

The only reason it works like normal in the same world is because the safe location method exempts locations in the same world as the teleporting player, and thus this check is never even performed in the first place for those players anyway.

**Demo**
```
[16:22:49 INFO]: CONSOLE issued server command: /ess version
[16:22:49 INFO]: Server version: 1.13.2-R0.1-SNAPSHOT git-Paper-"16db0e6a" (MC: 1.13.2)
[16:22:49 INFO]: EssentialsX version: 2.16.1.9
[16:22:49 INFO]: LuckPerms version: 4.3.73
[16:22:49 INFO]: Vault is not installed. Chat and permissions may not work.
```
https://streamable.com/71072

### CODE REVIEW

**Comment by mdcfe** [comment]
> I'm not exactly sure if adding water to `HOLLOW_MATERIALS` is the right decision, but it fixes the issue, and I personally don't really see any point in not having water in the list.

I'm not really sure what the rationale behind having two separate lists of materials was in the first place, to be honest. I think it was part of the initial 1.13 commit - @drtshock any ideas/thoughts?

**Comment by pop4959** [comment]
TLDR; make this a configuration option

I can see the usefulness of this, since oftentimes players will want to teleport back into water for various reasons, especially in 1.13, since water has become much more hospitable. You can use water breathing enchantments, sponges, or even conduit power to live much better underwater. Swimming makes travelling underwater even easier, and there is much more to explore such as the dungeons, ships, and coral reef. Some players (even on my small server) wish to build underwater houses.

However, obviously teleportation into water can also be dangerous. In many servers teleporting can be more of a luxury, in addition to the kinds of mobility people have. This issue should consider probably consider that, since Essentials is not built for any one server type. To prevent the possibility of this impacting servers negatively, and causing players to down, it might be good for it to be a configuration option.

I know personally, I would love to enable this on my server. I've had some players complain about not being able to teleport into water, since essentials safety ends up teleporting them pretty far away from where they meant to be.

Opinions?

**Comment by mdcfe** [comment]
@pop4959 I agree with making this configurable.

**Comment by caojohnny** [comment]
2 other maintainers support your opinion so I will get to this when I have time, poke me if I don't get around to it in the next few days or so

**Comment by mdcfe** [comment]
Closed in favour of #2520.
